cmd/bbolt/command_surgery_meta.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"go.etcd.io/bbolt/internal/common"
+)
+
+func newSurgeryMetaCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "meta <subcommand>",
+		Short: "meta page related surgery commands",
+	}
+
+	cmd.AddCommand(newSurgeryMetaValidateCommand())
+	cmd.AddCommand(newSurgeryMetaUpdateCommand())
+
+	return cmd
+}
+
+func newSurgeryMetaValidateCommand() *cobra.Command {
+	metaValidateCmd := &cobra.Command{
+		Use:   "validate <bbolt-file> [options]",
+		Short: "Validate both meta pages",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("db file path not provided")
+			}
+			if len(args) > 1 {
+				return errors.New("too many arguments")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return surgeryMetaValidateFunc(args[0])
+		},
+	}
+	return metaValidateCmd
+}
+
+func surgeryMetaValidateFunc(srcDBPath string) error {
+	if _, err := checkSourceDBPath(srcDBPath); err != nil {
+		return err
+	}
+
+	var pageSize uint32
```

Comment:
Why `readMetaPage` function above uses `guts_cli.ReadPageAndHWMSize` function to get meta page size while, this function passes 0 on the first iteration?

Reply:
The `pageSize` is used to locate the second meta page. The first meta page always start from offset 0.

Reply:
Ok, but why `readMetaPage` function is different?

Reply:
I get that, I didn't ask what's the difference between them. I asked why one uses `guts_cli.ReadPageAndHWMSize` to get page size, while other doesn't.

Reply:
`readMetaPage` needs to read the active meta page, which might be 0 or 1. If it's 1, then it needs to know the pageSize. It's the reason why it depends on `ReadPageAndHWMSize`.

`ReadMetaPageAt` already has an input parameter `pageSize`.

```
ReadMetaPageAt(dbPath string, metaPageId uint32, pageSize uint32)
```